test/loader/serializeTypeTest.cpp

```diff
@@ -361,4 +361,182 @@ TEST(serializeTypeTest, SerializeFuncRefType) {
   };
   EXPECT_EQ(Output, Expected);
 }
+
+TEST(serializeTypeTest, SerializeGCType) {
```

Comment:
I think you can combine this with the `SerializeFuncRefType` test, and rename as `SerializeValType`.
After all they are all tests about serializing the value types.
Just take care about the proposal configuration.